### DISCUSSION TITLE
Update with packageRules

### DISCUSSION THREAD

**Discussion by ram-you** [renovatebot/renovate#23853] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

Hi,facing issue with "antfu/icones" repository. 
Configuration file: [https://github.com/antfu/icones/blob/main/.github/renovate.json]()
Despite meeting the update criteria, the "@iconify/json" package version remains unchanged.
I would greatly appreciate your assistance.

### Relevant debug logs

```
{
  "automerge": false,
  "rangeStrategy": "replace",
  "extends": [
    "config:recommended",
    "group:allNonMajor",
    ":preserveSemverRanges"
  ],
  "packageRules": [
    {
      "matchUpdateTypes": ["minor", "patch", "pin", "digest"],
      "matchPackagePatterns": ["^@iconify/json"],
      "extends": ["schedule:daily"],
      "minimumReleaseAge": null,
      "automerge": false,
      "matchDatasources": ["npm"]
    },
    {
      "matchUpdateTypes": ["minor", "patch"],
      "excludePackagePatterns": ["^@iconify/json"],
      "extends": ["schedule:weekly"]
    }
  ]
}
```

### Have you created a minimal reproduction repository?

Placeholder value, please select the correct response from the dropdown

**Comment by rarkins** [comment]
This is the expected behavior when rangeStrategy=replace and you are using ranges. In short, the range doesn't need changing so therefore there is no PR.

Until #21438 is supported, you would need to set rangeStrategy=pin so that first your ranges are converted to versions, and then after that you get PRs whenever a new version is available

**Reply by ram-you** [reply]
Hello @rarkins,

I greatly appreciate your assistance throughout this process. I've put together a new configuration, as shared below. Before I proceed with testing, I would like to seek your expert opinion on whether this configuration meets the desired standards. Your insights would be invaluable.
```
{
  "automerge": false,
  "rangeStrategy": "pin",
  "extends": [
    "config:recommended",
    "group:allNonMajor",
    ":preserveSemverRanges"
  ],
  "packageRules": [
    {
      "matchUpdateTypes": ["pin"],
      "matchPackagePatterns": ["^@iconify/json"],
      "extends": ["schedule:daily"] 
    },
    {
      "matchUpdateTypes": ["minor", "patch"],
      "excludePackagePatterns": ["^@iconify/json"],
      "extends": ["schedule:weekly"]
    }
  ]
}

```
Thank you once again for your support and guidance.

Best regards,

**Reply by rarkins** [reply]
Hi, I do have some recommended changes but first need to ask some questions. It looks like the `@iconify/json` package needs to be treated specially. Can you describe that in simple terms? e.g. you would like for it to be updated separately from other packages? And you'd like it to be updated daily while other packages are updated weekly?

**Reply by ram-you** [reply]
Absolutely! The @iconify/json package holds a distinctive role within our project, warranting a distinct approach to updates compared to other packages.

Our primary goal is to maintain a daily update cadence for the @iconify/json package. This frequency aligns with the dynamic nature of the open-source vector icons it manages. Keeping up with daily updates ensures that we seamlessly incorporate the most recent additions and improvements to our icon sets.

In contrast, for the remaining packages, we find a weekly update frequency to be well-suited. This balance ensures we stay current with our core dependencies while giving special attention to the @iconify/json package.

However, if feasible, we would appreciate the ability to enable automated updates for the @iconify/json package.

Best regards,

**Reply by rarkins** [reply]
Here's my recommended config:

```json
{
  "extends": [
    "config:recommended"
  ],
  "rangeStrategy": "pin",
  "packageRules": [
    {
      "description": "Group all non-major updates weekly except @iconify/json",
      "extends": ["schedule:weekly"],
      "matchPackagePatterns": ["*"],
      "excludePackageNames": ["@iconify/json"],
      "matchUpdateTypes": ["minor", "patch"],
      "groupName": "all non-major dependencies",
      "groupSlug": "all-minor-patch"
    },
    {
      "description": "Create non-major @iconify/json updates daily",
      "matchUpdateTypes": ["minor", "patch"],
      "matchPackagePatterns": ["^@iconify/json"],
      "extends": ["schedule:daily"] 
    },
    {
      "description": "Suppress major updates using Dependency Dashboard",
      "matchUpdateTypes": ["major"],
      "dependencyDashboardApproval": true
    }
  ]
}
```

The result of this will be that you will get a one-time PR pinning all dependencies in `package.json` to specific versions rather than ranges. After that you will get regular "replace" updates updating from one version to another.

**Reply by ram-you** [reply]
Hello @rarkins,

In consideration of the "@iconify/json" package's daily updates, we're contemplating the idea of enabling automated updates by adding **"automerge": true** to the configuration:
```
{
  "description": "Create non-major @iconify/json updates daily",
  "matchUpdateTypes": ["minor", "patch"],
  "matchPackagePatterns": ["^@iconify/json"],
  "extends": ["schedule:daily"],
  "automerge": true
}

```
We're curious to gather your insights on this approach. Given the daily update frequency, are there any potential downsides or risks we should consider before enabling auto merge for this package?

**Reply by rarkins** [reply]
Ideally you have GitHub's automerge feature enabled on your repo, in that case Renovate will enable GitHub automerge at time of PR creation and it will automerge itself once tests pass

**Comment by ram-you** [comment]
Thank you very much for your insightful recommendations and the well-structured configuration you've provided. We truly appreciate your guidance in refining our Renovate setup.

We're excited to implement the configuration you've suggested and give it a try in our project. Your expertise and contributions to the Renovate community are invaluable, and we're grateful for your support.

Once we've had a chance to test the new configuration, we'll provide feedback on how it performs in our specific environment. Your input has been immensely helpful, and we look forward to optimizing our version management and updates accordingly.

Just a quick update: We've submitted a pull request to the "/antfu/icones" repository with the discussed configuration changes. You can find it here: [PR #189](https://github.com/antfu/icones/pull/189).

Thank you once again for your time and expertise.

Best regards,
